core/src/main/java/io/grpc/StatusRuntimeException.java

```diff
@@ -66,4 +68,27 @@ public final Status getStatus() {
   public final Metadata getTrailers() {
     return trailers;
   }
+
+  /**
+   * Wraps the given {@link Throwable} in a {@link StatusRuntimeException}. If it contains an
+   * embedded {@link StatusException} or {@link StatusRuntimeException}, the returned exception will
+   * contain the embedded trailers and status, with the given exception as the cause. Otherwise, an
+   * exception will be generated from an {@link Status#UNKNOWN} status.
+   */
+  @ExperimentalApi
+  public static StatusRuntimeException fromThrowable(Throwable t) {
```

Comment:
If it were package-private in client-calls that would be fine. I'd also be okay if it was in `internal` in `GrpcUtil`. I'm not bothered too much where, just as long as it does not appear at all like a public interface (even if unstable). So I guess even here would be fine if it were `@Internal`, but it is probably better to be more out-of-sight of our users (which both `internal` and package-private do).